### PR TITLE
dev-java/jdbc-postgresql: Version bump

### PR DESCRIPTION
**Pull request by a17r**
Test suite built successfully; new test dependency dev-java/xml-commons, sigh

Changelog:
https://jdbc.postgresql.org/documentation/changelog.html#version_9.4-1204

Package-Manager: portage-2.2.23

### CODE REVIEW

**Comment by mgorny** [comment]
@gentoo/java

**Comment by mgorny** [comment]
@a17r, up to some un-screwing up this package or should I focus on diff? ;-P

**Comment by a17r** [comment]
oh if you ask me, focus on diff :p

**Comment by mgorny** [comment]
Hmm, git log doesn't want to catch a diff, sorry :-P. Plus you didn't GPG-sign it, so I'm going to nitpick now!

File: dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		dev-java/junit:4
+		dev-java/xml-commons
+	)"
+RDEPEND=">=virtual/jre-1.6"
+
+S="${WORKDIR}/postgresql-jdbc-${MY_PV}.src"
+
+java_prepare() {
+	# Strip build.xml of maven deps
+	sed -i -e '/<classpath.*dependency\.compile\.classpath/c\' build.xml || die
```
Comment (mgorny) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: This really cries for collapsing into one `sed` with multiple `-e`s.

File: dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+EANT_DOC_TARGET="publicapi"
+
+src_compile() {
+	EANT_BUILD_TARGET="release-version jar"
+	java-pkg-2_src_compile
+
+	# There is a task that creates this doc but I didn't find a way how to use system catalog
+	# to lookup the stylesheet so the 'doc' target is rewritten here to use system call instead.
+	if use doc; then
+		mkdir -p "${S}/build/doc"
```
Comment (mgorny) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: ||die

File: dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		xsltproc -o "${S}/build/doc/pgjdbc.html" http://docbook.sourceforge.net/release/xsl/current/xhtml/docbook.xsl \
```
Comment (mgorny) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: ||die again

File: dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+src_test() {
+	einfo "In order to run the tests successfully, you have to have:"
+	einfo "1) PostgreSQL server running"
+	einfo "2) database 'test' defined with user 'test' with password 'test'"
+	einfo "   as owner of the database"
+	einfo "3) plpgsql support in the 'test' database"
+	einfo
+	einfo "You can find a general info on how to perform these steps at"
+	einfo "https://wiki.gentoo.org/wiki/PostgreSQL"
```
Comment (mgorny) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: @gentoo/java this is crazy and you know that. Please set RESTRICT=test before it messes up someone's postgres.

Comment (mgorny) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: Also it won't work with fancy sandboxes in place.

File: dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+	ANT_TASKS="ant-junit" eant test -Dgentoo.classpath=$(java-pkg_getjars --build-only "junit-4,xml-commons")
+}
+
+src_install() {
+	java-pkg_newjar build/jars/postgresql*.jar jdbc-postgresql.jar
+
+	if use doc ; then
+		java-pkg_dojavadoc build/publicapi
+		dohtml build/doc/pgjdbc.html
```
Comment (mgorny) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: Suggest docinto+dodoc :-P.

**Comment by mgorny** [comment]
I'd say RESTRICT=test is important, ||die quite important, rest unimportant.

File: dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild [hunk 6]
```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="5"
+
+JAVA_PKG_IUSE="doc source"
+
+inherit java-pkg-2 java-ant-2
```
Comment (hasufell) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: `eutils` inherit missing, you use epatch in line 51

Comment (mgorny) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: Did someone forget to run `repoman full` then? I thought it catches stuff like that these days.

Comment (a17r) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: Apparently no, it wasn't caught

Comment (hasufell) [inline dev-java/jdbc-postgresql/jdbc-postgresql-9.4_p1204.ebuild]: maybe it only catches it if epatch is not a defined function? It's implicitly inherited via java-utils-2

**Comment by monsieurp** [comment]
The failure is due to a package I masked and later on unmasked. It has nothing to do with a bug with the ebuild itself or the PR. @a17r I'm going to merge it.

**Pull request merged**